server/handler/github_event_handler.go

```diff
@@ -0,0 +1,106 @@
+	ClientCreator     githubapp.ClientCreator
+	GithubEvents      []string
+	EventContextStore store.EventContextStore
+}
+
+func (h *githubEventHandler) Handles() []string {
+	return h.GithubEvents
+}
+
+func (h *githubEventHandler) Handle(ctx context.Context, eventType, deliveryID string, payload []byte) error {
+	log.Info(logEventType).Msgf("%s is received!", eventType)
+	eventContext, err := model.ConvertPayloadToEventContext(eventType, payload)
+	if err != nil {
+		log.Info(logEventType).Msgf("Error occurred while deserializing request: %v", err)
+		return errors.Wrap(err, "failed to parse event payload")
+	}
+
+	eventContext.Log()
+
+	pipeline := model.GetTargetPipeline(eventContext, h.Config.Pipelines)
+
+	if pipeline != nil {
+		if err := h.triggerPipeline(ctx, eventContext, *pipeline); err != nil {
+			log.Info(logEventType).Msgf("Error occurred while triggering pipeline request: %v", err)
+			return errors.Wrap(err, "failed to trigger pipeline")
+		}
+	} else {
+		log.Info(logEventType).
+			Str("type", eventType).
+			Str("workflow", eventContext.GetWorkflow()).
+			Str("repository", eventContext.GetRepository()).
+			Str("reference", eventContext.GetReference()).
+			Msg("No pipeline configured")
+	}
+	return nil
+}
+
+func newGithubEventHandler(cc githubapp.ClientCreator, config *config.Config, eventContextStore store.EventContextStore) *githubEventHandler {
+	return &githubEventHandler{
+		ClientCreator:     cc,
+		Config:            config,
+		GithubEvents:      []string{"push", "workflow_run"},
+		EventContextStore: eventContextStore,
+	}
+}
+
+func (h *githubEventHandler) triggerPipeline(ctx context.Context, eventContext model.EventContext, pipeline config.PipelineConfig) error {
+	log.Info("trigger").
+		Str("org", pipeline.Organization).
+		Str("repo", pipeline.Repository).
+		Str("workflow", pipeline.Workflow).
+		Msg("Will Trigger Pipeline!")
+	client, err := h.ClientCreator.NewInstallationClient(eventContext.GetInstallationID())
+
+	if err != nil {
+		return err
+	}
+	token := h.EventContextStore.Store(eventContext)
+	inputs := map[string]interface{}{
+		"repository":    eventContext.GetRepository(),
+		"name":          eventContext.GetName(),
+		"workflow":      eventContext.GetWorkflow(),
+		"workflowRunId": strconv.FormatInt(eventContext.GetWorkflowRunID(), 10),
+		"commmitHash":   eventContext.GetCommitHash(),
+		"ref":           eventContext.GetReference(),
+		"fork":          strconv.FormatBool(eventContext.IsFork()),
+		"type":          "pr",
+		"botToken":      token,
+		"botBaseUrl":    h.Config.Server.BaseURL,
+	}
+	if eventContext.IsTag() {
+		inputs["type"] = "tag"
+	}
+	if eventContext.IsBranch() {
+		inputs["type"] = "branch"
+	}
+	deRequest := github.CreateWorkflowDispatchEventRequest{
+		Ref:    "main",
+		Inputs: inputs,
+	}
+	_, err = client.Actions.CreateWorkflowDispatchEventByFileName(ctx, pipeline.Organization, pipeline.Repository, pipeline.Workflow, deRequest)
+	if err != nil {
+		return err
```

Comment:
I forgot to mention above but when there is a call to a different package etc. it's good to make `errors.Wrap()` for better stacktraces. a great post by Dave https://dave.cheney.net/tag/error-handling

Reply:
Errors are wrapped.